scratch/benchmarks/remote.py

```diff
@@ -164,15 +164,14 @@ async def run_tasks(
         async with asyncio.timeout(timeout):
             for i in range(n):
                 # Wait until either the port is ready or the process dies
-                tasks = [
-                    asyncio.create_task(wait_port(server, port_base + i)),
-                    asyncio.create_task(procs[i].wait(check=True)),
-                ]
-                done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
-                for task in pending:
-                    task.cancel()
-                for task in done:
-                    task.result()  # Re-raise any exceptions
+                async with asyncio.TaskGroup() as tg:
+                    tasks = [
+                        tg.create_task(wait_port(server, port_base + i)),
+                        tg.create_task(procs[i].wait(check=True)),
+                    ]
+                    done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
+                    for task in pending:
+                        task.cancel()
```

Comment:
Ah, I gather the move to using a `TaskGroup` here is to avoid the old Line 175 of re-raising any exceptions?

Reply:
I was generally trying to use TaskGroups where they fit. But yes, it's an advantage of TaskGroups. It will also better handle the case where multiple completed tasks raised exceptions (they'll all get bundled into an ExceptionGroup, while the old code would just raise the first one and drop the rest on the floor). I also realised that the old code doesn't actually wait for the cancelled tasks to complete (which might not be immediately if they have `final` code).

Reply:
> I also realised that the old code doesn't actually wait for the cancelled tasks to complete

Aha, it simply executed `task.cancel()` and moved on with life. Not to drag on, but out of interest
* Here we/you seem to be doing a similar thing (`task.cancel()`), but
* Now wrapped with an `async with asyncio.TaskGroup()`

How does the TaskGroup context help "resolve" incomplete cancelled tasks?

Reply:
Ah, reading your reply on my comment regarding `pytest.raises`, I'm thinking
> the context manager swallows any `CancelledError`s.

Reply:
Yes. It also waits for all tasks in the task group to complete.